x-pack/plugins/stack_connectors/common/inference/schema.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+export const ConfigSchema = schema.object({
+  provider: schema.string(),
+  taskType: schema.string(),
+  inferenceId: schema.string(),
+  providerConfig: schema.object({}, { unknowns: 'allow', defaultValue: {} }),
+  taskTypeConfig: schema.object({}, { unknowns: 'allow', defaultValue: {} }),
+});
+
+export const SecretsSchema = schema.object({
+  providerSecrets: schema.object({}, { unknowns: 'allow', defaultValue: {} }),
+});
+
+export const ChatCompleteParamsSchema = schema.object({
+  input: schema.string(),
+});
+
+export const ChatCompleteResponseSchema = schema.object({
+  completion: schema.arrayOf(
+    schema.object({
+      result: schema.string(),
+    }),
+    { defaultValue: [] }
+  ),
+});
+
+export const RerankParamsSchema = schema.object({
+  input: schema.arrayOf(schema.string(), { defaultValue: [] }),
+  query: schema.string(),
+});
+
+export const RerankResponseSchema = schema.object({
+  rerank: schema.arrayOf(
+    schema.object({
+      text: schema.string(),
+      index: schema.number(),
+      score: schema.number(),
+    }),
+    { defaultValue: [] }
+  ),
+});
+
+export const SparseEmbeddingParamsSchema = schema.object({
+  input: schema.string(),
+});
+
+export const SparseEmbeddingResponseSchema = schema.object({
+  sparse_embedding: schema.arrayOf(schema.object({}, { unknowns: 'allow' }), { defaultValue: [] }),
```

Comment:
The `unknowns: 'allow'` on params / responses (vs `config` or `secrets`) should be fine - given my comment above about usage in `config` and `secrets`.